### PR TITLE
Exporting comments in CSV

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**

### CODE REVIEW

File: client/src/app/core/ui-services/csv-export.service.ts [hunk 1]
```diff
@@ -164,20 +151,17 @@ export class CsvExportService {
...
-    public checkCsvTextSafety(input: string, tsList: string[]): string[] {
-        if (input === null || input === undefined) {
-            return tsList;
+    public checkCsvTextSafety(input: string): string {
+        if (!input) {
+            return '';
         }
-
-        return tsList.filter(char => input.indexOf(char) < 0);
+        return '"' + input.replace(/"/g, '""').replace(/(\r\n\t|\n|\r\t)/gm, ".") + '"';
```
Comment (MaximilianKrambach) [inline client/src/app/core/ui-services/csv-export.service.ts]: Okay, going the way of changing the exported strings. Are the more complicated ones (e.g. with quotes, html formatting, special characters, linebreaks) still save for re-import? (That's the most common use case I heard about)

Comment (GabrielInTheWorld) [reply client/src/app/core/ui-services/csv-export.service.ts]: Yes, I've tested this by trying re-importing the motions exported before.

File: client/src/app/core/ui-services/csv-export.service.ts [hunk 2]
```diff
@@ -112,15 +108,15 @@ export class CsvExportService {
                 label = column.label;
             }
             label = this.capitalizeTranslate(label);
-            tsList = this.checkCsvTextSafety(label, tsList);
+            // tsList = this.checkCsvTextSafety(label, tsList);
```
Comment (tsiegleauq) [inline client/src/app/core/ui-services/csv-export.service.ts]: please remove

File: client/src/app/site/motions/services/motion-csv-export.service.ts [hunk 3]
```diff
@@ -83,7 +87,12 @@ export class MotionCsvExportService {
      * @param contentToExport content properties to export
      * @param crMode
      */
-    public exportMotionList(motions: ViewMotion[], contentToExport: string[], crMode?: ChangeRecoMode): void {
+    public exportMotionList(
+        motions: ViewMotion[],
+        contentToExport: string[],
+        crMode?: ChangeRecoMode,
+        comments?: number[]
```
Comment (tsiegleauq) [inline client/src/app/site/motions/services/motion-csv-export.service.ts]: both `crMode` and `comments` should be part of `contentToExport`.

If you don't want to fix this now, add a TODO

File: client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.ts [hunk 4]
```diff
@@ -323,7 +323,8 @@ export class MotionListComponent extends BaseListViewComponent<ViewMotion> imple
                     this.motionCsvExport.exportMotionList(
                         data,
                         [...exportInfo.content, ...exportInfo.metaInfo],
-                        exportInfo.crMode
+                        exportInfo.crMode,
```
Comment (tsiegleauq) [inline client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.ts]: as before

**Pull request merged**